### PR TITLE
Created GCS link viewer/downloader

### PR DESCRIPTION
**Pull request by MattPutnam**
Also commonized some things as I needed them:
* Created common util for formatting dates, used where possible
* Made "X" button a component and updated all usages

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs [hunk 1]
```diff
@@ -3,8 +3,10 @@
     [clojure.string :refer [blank?]]
     [dmohs.react :as react]
     [org.broadinstitute.firecloud-ui.common :as common]
+    [org.broadinstitute.firecloud-ui.endpoints :as endpoints]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs]: Alphabetize.

File: src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs [hunk 2]
```diff
@@ -359,3 +368,82 @@
...
+      (when (or (:show-dialog? @state) (:loading? @state))
+        (let [{:keys [data error]} (:response @state)
+              data-size (when data (data "size"))
+              labeled (fn [label content]
+                        [:div {}
+                         [:div {:style {:display "inline-block" :width 120}} (str label ": ")]
+                         [:span {} content]])]
+          [:div {:style {:position "fixed" :top 0 :left 0 :right 0 :bottom 0 :zIndex 9999
+                         :fontSize "initial" :fontWeight "initial"}}
+           [Dialog
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs]: Would it be possible to use OkCancel dialog?

File: src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs [hunk 3]
```diff
@@ -359,3 +368,82 @@
       (if (blank? (entity "documentation"))
         [:div {:style {:fontStyle "italic" :fontSize "90%"}} "No documentation provided"]
         [:div {:style {:fontSize "90%"}} (entity "documentation")])])})
+
+(react/defc GCSFilePreviewLink
+  {:render
+   (fn [{:keys [props state refs this]}]
+     (assert (:bucket-name props) "No bucket name provided")
+     (assert (:object props) "No GCS object provided")
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs]: These asserts are awesome. Good documentation.

**Comment by dmohs** [comment]
:+1: I like the idea of extracting the XButton, but we do have a story to migrate stuff to the OkCancel dialog. Wouldn't that make things more consistent throughout the UI?

Other comments very minor. Merge when ready.

File: src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs (hunk 2 above)
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs]: I put it in the OKCancelForm and it looks good, but now you can't right-click to rename the file.  I had to go back to the dummy anchor that the button programmatically clicks.

**Pull request merged**